### DISCUSSION TITLE
Are states cleared when a workflow is removed from the "stack"?

### DISCUSSION THREAD

**Discussion by zoltish** [square/workflow-kotlin#528] [General] [answered]
If workflow A renders children X and Y, then stops rendering Y and later on resumes doing so (with the same key); does Y retain its state or is that destroyed when the workflow "is destroyed" and as such initialized again?

**Comment by rjrjr** [comment, marked as answer]
The latter, Y's state is lost when you finish a `render()` call without referencing it, and you start again from scratch the next time.

It's not uncommon for a parent to keep a child workflow "running in the background" by continuing to render the child w/o incorporating its rendering — workflow sessions are cheap. We've also sometimes had a child emit a check point object as output, which the parent can then provide via props to resume work.

**Reply by zoltish** [reply]
That makes sense: no need to keep the resources around if they might not be used again - or be stale at that point. Thanks for responding! :)
